orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -182,11 +182,11 @@ Status OrtSaveInternal(
     const std::vector<ONNX_NAMESPACE::TensorProto>& non_trainable_tensor_protos,
     const PathString& checkpoint_path) {
   // Make sure name unique across trainable and non-trainable lists.
-  std::unordered_set<std::string> trainable_unique_names;
-  std::unordered_set<std::string> non_trainable_unique_names;
+  std::set<std::string> trainable_unique_names;
```

Comment:
curious why is this change required?

Reply:
The problem is that we perform `set_intersection` on the `unordered_set`. The definition from https://en.cppreference.com/w/cpp/algorithm/set_intersection

> Constructs a sorted range beginning at d_first consisting of elements that are found in both sorted ranges [first1, last1) and [first2, last2).

Is expecting the inputs to be sorted. While the linux build succeeds, the windows build throws an exception at runtime:

> Assertion failed: sequence not ordered

More details [here](https://stackoverflow.com/questions/48158811/c-library-method-for-intersection-of-two-unordered-set/48158882#48158882).

To circumvent this, we can either loop over the `unordered_set` manually, or use `set` instead. I chose the latter since this is a one-time expense to find the intersection.